File: android/app/src/main/java/jagtheora/ogg/OggStreamState.java
package jagtheora.ogg;

import jagtheora.misc.SimplePeer;

public final class OggStreamState extends SimplePeer {

    public OggStreamState(int arg0) {
        if (!this.init(arg0)) {
            throw new IllegalStateException();
        }
    }

    public boolean init(int arg0) { return false; }

    public int packetOut() { return 0; }

    public boolean isEOS() { return false; }

    public int packetPeek(OggPacket arg0) { return 0; }

    protected void clear() {}

    public boolean reset() { return false; }

    public int packetOut(OggPacket arg0) { return 0; }

    public int packetPeek() { return 0; }

    public boolean resetSerialNo(int arg0) { return false; }

    public boolean pageIn(OggPage arg0) { return false; }
}

File: src/jvmMain/kotlin/jagtheora/ogg/OggStreamState.kt
package jagtheora.ogg

import jagtheora.misc.SimplePeer

actual class OggStreamState actual constructor(arg0: Int) : SimplePeer() {
    init {
        check(this.init(arg0))
    }

    actual external fun packetPeek(arg0: OggPacket?): Int

    /*private*/ actual external fun packetPeek(): Int

    /*private*/ actual external fun init(arg0: Int): Boolean

    /*private*/ actual val isEOS: Boolean
        external get

    actual external override fun clear()

    actual external fun pageIn(arg0: OggPage?): Boolean

    /*private*/ actual external fun reset(): Boolean

    actual external fun packetOut(): Int

    actual external fun packetOut(arg0: OggPacket?): Int

    /*private*/ actual external fun resetSerialNo(arg0: Int): Boolean
}
